lightningd/notification.c

```diff
@@ -0,0 +1,35 @@
+#include "lightningd/notification.h"
+#include <ccan/array_size/array_size.h>
+
+const char *notification_topics[] = {
+	"connect",
+	"disconnect",
+};
+
+bool notifications_have_topic(const char *topic)
+{
+	for (size_t i=0; ARRAY_SIZE(notification_topics); i++)
+		if (streq(topic, notification_topics[i]))
+			return true;
+	return false;
+}
+
+void notify_connect(struct lightningd *ld, struct pubkey *nodeid,
```

Comment:
so just for the gist I could create a method `void notify_incoming_htlc(struct lightningd *ld, struct pubkey *nodeid, struct channel *chan, struct htlc *h)` (not sure if `struct channel` and `struct htlc` exist) and create a json string and send it via plugin_notify to the plugin.  I would obviously have to call that function at the place where incoming htlc's are being handled. But other than that I would not have to extend other parts of c-lightning or am I missing something?

Reply:
Yes, the idea is that you'd create a `notify_xyz` function that gathers all the information and formats it into a JSON message and then passes that to `plugins_notify` which will take care of dispatching notifications to all subscribers.